test/tests/functional/pbs_pbsnodes.py

```diff
@@ -0,0 +1,139 @@
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestPbsnodes(TestFunctional):
+
+    def setUp(self):
+        self.server.manager(MGR_CMD_DELETE, NODE, id="",
+                            sudo=True, expect=True)
+        self.server.manager(MGR_CMD_CREATE, NODE, id=self.mom.shortname)
+        self.server.expect(NODE, {'state': 'free'})
+
+    def expect_pbsnodes_val(self, user):
+        """
+        return expected value of pbsnodes
+        """
+
+        expect_dict = {}
+        host_short_name = self.mom.shortname
+        host_name = self.mom.hostname
+        pbs_version = self.server.pbs_version
+
+        ncpus_val = str(os.sysconf("SC_NPROCESSORS_ONLN"))
+        avail_mem = os.sysconf("SC_PAGE_SIZE")*os.sysconf("SC_PHYS_PAGES")/1024
+        avail_mem = str(avail_mem) + 'kb'
+
+        expect_dict['Mom'] = host_name
+        expect_dict['ntype'] = 'PBS'
+        expect_dict[ATTR_NODE_state] = 'free'
+        expect_dict['resources_available.vnode'] = host_short_name
+        expect_dict['resources_available.host'] = host_short_name
+        expect_dict['resources_assigned.accelerator_memory'] = '0kb'
+        expect_dict['resources_assigned.hbmem'] = '0kb'
+        expect_dict['resources_assigned.mem'] = '0kb'
+        expect_dict['resources_assigned.naccelerators'] = '0'
+        expect_dict['resources_assigned.ncpus'] = '0'
+        expect_dict['resources_assigned.vmem'] = '0kb'
+        expect_dict['resv_enable'] = 'True'
+        expect_dict['sharing'] = 'default_shared'
+        expect_dict['resources_available.mem'] = avail_mem
+        expect_dict['resources_available.ncpus'] = ncpus_val
+        expect_dict['pcpus'] = ncpus_val
+
+        platform = self.du.get_platform(hostname=self.mom.shortname)
+        if platform.startswith('linux'):
+            expect_dict['resources_available.arch'] = 'linux'
+
+        if user == 'root':
+            expect_dict['pbs_version'] = pbs_version
+            expect_dict['Port'] = '15002'
+
+        return expect_dict
+
+    def test_pbsuser_as_user(self):
+        """
+        validate pbsnodes default attribute for normal user
+        """
+
+        attr_dict = {}
+        out = self.expect_pbsnodes_val(TEST_USER)
+        command = os.path.join(self.server.pbs_conf['PBS_EXEC'],
+                               'bin', 'pbsnodes -a')
+        ret = self.du.run_cmd(self.server.hostname, command)
+        attr_list = ret['out']
+        list_len = len(attr_list) - 1
+        attr_list = filter(None, attr_list)
+        for i in range(1, list_len):
```

Comment:
Why are you starting from index 1?

Reply:
in ret['out'] first element is node name. to split attributes taking starting index 1.